src/components/chat/ui/ChatInput.tsx

```diff
@@ -28,14 +31,23 @@ export const ChatInput = () => {
         messageInputRef={ref}
         currentChannel={channel as any}
         messages={dataSource.messages}
-        sendUserMessage={(params) => {
-          const processedParams = handlers.onBeforeSendMessage(params);
-          dataSource.sendUserMessage(processedParams, handlers.onAfterSendMessage).then(handlers.onAfterSendMessage);
+        sendUserMessage={async (params) => {
+          const processedParams = await handlers.onBeforeSendMessage(params);
+          const message = await dataSource.sendUserMessage(processedParams, () => handlers.onAfterSendMessage());
+          handlers.onAfterSendMessage();
```

Comment:
It seems onAfterSendMessage being called onPending and onSuccess/onFailed. However, onAfterSendMessage() does not take message as input. How can it distinguish pending and success?

Reply:
Here’s how the logic works.

```ts
await dataSource.sendUserMessage(processedParams, () => handlers.onAfterSendMessage()); // on pending
handlers.onAfterSendMessage(); // after succeeded
```

Reply:
I understood that. I meant to ask `onAfterSendMessage: () => void;` should be `onAfterSendMessage: (message: BaseMessage) => void;`. Otherwise, onAfterSendMessage will not know when it is being called (pending or success/fail).

Reply:
Ah, right now it's only handling scrolling. That's why I didn't set it up to receive messages via a callback. It should be fine to add that later if needed.